lib/matplotlib/axes/_axes.py

```diff
@@ -4272,9 +4269,8 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xshape, yshape,
                 if not valid_shape:  # but at least one conversion succeeded.
                     raise ValueError(
                         "'c' argument has {nc} elements, which is not "
-                        "acceptable for use with 'x' with size {xs}, "
-                        "'y' with size {ys}."
-                            .format(nc=n_elem, xs=xsize, ys=ysize)
+                        "acceptable for use with 'x' and 'y' with size {xs}."
+                            .format(nc=csize, xs=xsize)
```

Comment:
just make this a f-string, or at least use the same names in the braces and outside (yes, I know, the problem was already there before)

Reply:
done